### DISCUSSION TITLE
The Emscripten Module Object

### DISCUSSION THREAD

**Discussion by Grey41** [pyodide/pyodide#3589] [Q&A] [answered]
Hello, I'd like to set the `canvas` attribute of the Emscripten `Module` object.
So far my JavaScript code looks like this.

```js
async function start() {
    const py = await loadPyodide()

    await py.loadPackage("glfw_package")
    py.runPython(`some code`)
}

start()
```

How can I pass my html canvas to Emscripten?

**Comment by hoodmane** [comment, marked as answer]
The Emscripten module is at `pyodide._module`. We offer no stability guarantees if you touch it, though there aren't too often changes. We'll have to work on a stable API for this.

**Reply by ryanking13** [reply]
I am thinking of adding public APIs to modify Module.canvas in #3508. I am working on other PRs right now, but hopefully I'll continue to work on that soon.
